pkg/webhook/webhook_test.go

```diff
@@ -356,83 +373,308 @@ func TestGitHubWrongSecret(t *testing.T) {
-		namespace: "default",
+		// when no secret is defined we accept the payload
+		"no-global-secret-no-gitrepo-secret-secret-in-gitrepo": {
+			secretValueInWebhook: "supersecretvalue",
+			globalSecret:         false,
+			gitrepoSecret:        false,
+			globalSecretKey:      "",
+			globalSecretValue:    "",
+			gitrepoSecretKey:     "",
+			gitrepoSecretValue:   "",
+			setSecretInGitrepo:   true,
+			expectedResCode:      http.StatusInternalServerError,
+			expectedCommitUpdate: false,
+		},
 	}
+	for _, tt := range tests {
+		ctlr := gomock.NewController(t)
+		mockClient := mocks.NewMockClient(ctlr)
+
+		gitRepo := &v1alpha1.GitRepo{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test",
+				Namespace: "gitrepoNamespace",
+			},
+			Spec: v1alpha1.GitRepoSpec{
+				Repo:   "https://github.com/example/repo",
+				Branch: "main",
+			},
+		}
 
-	w.github, _ = github.New(github.Options.Secret(""))
+		if tt.setSecretInGitrepo {
+			gitRepo.Spec.WebhookSecret = gitrepoSecretName
+		}
 
-	// we set only the values that we're going to use in the push event to make things simple
-	jsonBody := []byte(fmt.Sprintf(`
-	{
-	  "ref":"refs/heads/main",
-	  "after":"%s",
-	  "repository":{
-		"html_url":"https://github.com/example/repo"
-      }
-    }`, expectedCommit))
+		// List GitRepos mock call
+		mockClient.EXPECT().List(gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes().DoAndReturn(
+			func(ctx context.Context, list *v1alpha1.GitRepoList, opts ...client.ListOption) error {
+				list.Items = append(list.Items, *gitRepo)
+				return nil
+			},
+		)
+
+		// call for secret
+		mockClient.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(
+			func(ctx context.Context, name types.NamespacedName, secret *corev1.Secret, _ ...interface{}) error {
+				// check that we're calling Get with the expected name and Namespace
+				if tt.gitrepoSecret {
+					if name.Name != gitrepoSecretName {
+						t.Errorf("expecting calling secret Get with secret name %q, got %q", gitrepoSecretName, name.Name)
+					}
+					if name.Namespace != gitRepo.Namespace {
+						t.Errorf("expecting calling secret Get with secret namespace %q, got %q", gitRepo.Namespace, name.Namespace)
+					}
+				} else if tt.globalSecret {
+					// it expects the global secret name
+					if name.Name != webhookSecretName {
+						t.Errorf("expecting calling secret Get with secret name %q, got %q", webhookSecretName, name.Name)
+					}
+					// we're using "default" as the namespace for the webhook
+					if name.Namespace != "default" {
+						t.Errorf("expecting calling secret Get with secret namespace %q, got %q", "default", name.Namespace)
+					}
+				}
+				if tt.gitrepoSecret {
+					secret.Data = map[string][]byte{
+						tt.gitrepoSecretKey: []byte(tt.gitrepoSecretValue),
+					}
+					return nil
+				} else if tt.globalSecret {
+					secret.Data = map[string][]byte{
+						tt.globalSecretKey: []byte(tt.globalSecretValue),
+					}
+					return nil
+				}
```

Comment:
nit: why check those conditions again when we already have an if/else block over them above?

Reply:
The idea was to do the error checking first and the return of secret data later.
And, as this is a test file.. I thought it was better to do it that way to separate functionalities (the code is confusing enough if you are not familiar with it)

No strong opinion on any on these things, tbh.

Reply:
Understood. As you've noted, it's a test file, no worries.